classes/local/results.php

```diff
@@ -523,8 +523,9 @@ public static function diary_print_user_entry($context, $course, $diary, $user,
             echo '<td style="width:35px;">'.get_string('entry', 'diary').':</td><td>';
             echo userdate($entry->timecreated);
             // 20201202 Added link to show all entries for a single user.
-            $url = new moodle_url('mod/diary/reportsingle.php', ['id' => $id, 'user' => $user->id, 'action' => 'allentries']);
-            echo '  <a href="'.$url->out(false).'">'.get_string('reportsingle', 'diary')
+            echo '  <a href="reportsingle.php?id='.$id
+                .'&user='.$user->id
+                .'&action=allentries">'.get_string('reportsingle', 'diary')
```

Comment:
Not sure if I agree with changing from moodle_url to straight html. 
I would also recommend using the html_writer class to output html, instead of just string concatenation

Something like the following:

```
$url = new moodle_url('mod/diary/reportsingle.php', ['id' => $id, 'user' => $user->id, 'action' => 'allentries']);
echo html_writer::link($url, get_string('reportsingle', 'diary'))
```